### ISSUE TITLE
ValueError: could not convert string to float when computing shap_values for numpy.array input features type

### ISSUE THREAD

**Issue by amine-hef** [shap/shap#2208]
Hello,
I'm having this error when I try to generate shap_values for a LightGBM model that have categorical input features:
![image](https://user-images.githubusercontent.com/32578022/136035871-283cb4f5-1263-4ab8-93ff-9043449a8465.png)

Here you have a simple code to reproduce the warning:
```
df = pd.DataFrame({'A': ['Foo', 'Bar', 'Foo', 'Baz'], 'B': [1, 1, 0, 1], 'C': [1, 0, 1, 1]})
df['A'] = df['A'].astype('category')
lgb_train = lgb.Dataset(df[['A', 'B']], df['C'])

gbm = lgb.train({'objective': 'binary', 'min_data': 1, 'min_data_in_bin': 1},
                lgb_train,
                num_boost_round=1)

shap.initjs()
explainer = shap.TreeExplainer(gbm)
shap_values = explainer.shap_values(df[['A', 'B']].values)
```

my shap version: '0.39.0'
Also, it works if I provide directly the dataframe instead of the values but idk why then.

**Comment by seakelle** [comment]
If you change the last line of your example to `shap_values = explainer.shap_values(df[['A', 'B']])` (without the `.values`), it should work with the categorical feature.

However, the `__call__` function is still not working with categorical features (i.e. `explainer(df[['A', 'B']])` will fail): #2144

**Comment by detrin** [comment]
@thatlittleboy Candidate for closing?

**Comment by thatlittleboy** [comment]
Fixed by dsgibbons#88, and these changes were merged into the main repo and released in version 0.42.0. Should not be an issue anymore
